aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/CRTNative.kt

```diff
@@ -2,30 +2,53 @@
  * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
+
 package aws.sdk.kotlin.crt
 
+import kotlinx.atomicfu.atomic
+import kotlinx.cinterop.ExperimentalForeignApi
+import kotlinx.cinterop.convert
+import kotlinx.cinterop.staticCFunction
+import kotlinx.cinterop.toKString
+import libcrt.*
+import platform.posix.atexit
+
+@OptIn(ExperimentalForeignApi::class)
 public actual object CRT {
+    private val initialized = atomic(false)
+
     /**
      * Initialize the CRT libraries if needed
      */
-    public actual fun initRuntime(block: Config.() -> Unit) { }
+    public actual fun initRuntime(block: Config.() -> Unit) {
+        if (!initialized.compareAndSet(false, true)) return
+
+        val config = Config().apply(block)
+
+        // bootstrap our allocator
+        s_crt_kotlin_init_allocator(CrtDebug.traceLevel)
+
+        aws_common_library_init(Allocator.Default)
+        aws_compression_library_init(Allocator.Default)
+        aws_io_library_init(Allocator.Default)
+        aws_http_library_init(Allocator.Default)
+
+        Logging.initialize(config)
+        atexit(staticCFunction(::cleanup))
+    }
```

Comment:
**Correctness**: The atomic prevents multiple callers from simultaneously initializing the CRT runtime but doesn't prevent callers that take the early return from proceeding based on the assumption that the CRT _is_ initialized (which it may not be yet). Is that safe? Do we need a mutex here rather than an atomic check?

Reply:
Good observation. I can't remember why this was done this way but I can take a guess that it's because the configuration for CRT Java doesn't map cleanly here. They use [system properties ](https://github.com/awslabs/aws-crt-java/blob/main/src/main/java/software/amazon/awssdk/crt/Log.java#L22)to control config which won't work on native obviously. 

They also use a [static initializer](https://github.com/awslabs/aws-crt-java/blob/main/src/main/java/software/amazon/awssdk/crt/CRT.java#L56) which is what's giving them the only once init behavior. 

I see a few options:

1. Change this to a static initializer and remove/deprecate `Config` in favor of system properties on JVM and environment variables on native. 
3. Add a mutex

Reply:
I like the flexibility of `Config` so I'd be more in favor of a mutex.

Reply:
Using a mutex would make this a `suspend` function (as it probably should've been), which is a breaking change to the `CRT` interface. Are we ok with that?

Reply:
It doesn't have to be `suspend` it can just use runblocking (the purpose of which is to bridge suspend and non-suspend. In this case we want to expose a synchronous interface).